File: src/PythonClient/lib/quixstreaming/models/streampackage.py
from quixstreaming.models.netdict import NetDict
from quixstreaming.models.netlist import NetList
from quixstreaming import __importnet
from quixstreaming.helpers import *
from datetime import datetime
import Quix.Sdk.Process


class StreamPackage(object):
    """
        Default model implementation for non-typed message packages of the Process layer. It holds a value and its type.
    """

    @property
    def type(self) -> str:
        """Type of the content value"""
        return self.__wrapped.Type

    @property
    def value(self) -> object:
        """Content value of the package"""
        return self.__wrapped.Value

    @property
    def transport_context(self) -> NetDict:
        """Get the additional metadata for the stream"""
        return NetDict(self.__wrapped.TransportContext)

    def to_json(self) -> str:
        """
            Serialize the package into Json
        """
        return self.__wrapped.ToJson()
        

    def __init__(self, net_object: Quix.Sdk.Process.Models.StreamPackage):
        """
            Initializes a new instance of StreamPackage.

            NOTE: Do not initialize this class manually. Will be initialized by StreamReader.on_package_received
            Parameters:

            net_object (.net object): The .net object representing StreamPackage.
        """

        self.__wrapped: Quix.Sdk.Process.Models.StreamPackage = net_object

File: src/CsharpClient/QuixStreams.Telemetry/Models/Package/StreamPackage.cs
using System;
using Newtonsoft.Json;
using QuixStreams.Kafka;
using QuixStreams.Kafka.Transport;

namespace QuixStreams.Telemetry.Models
{
    /// <summary>
    /// Default model implementation for non-typed message packages of the Telemetry layer. It holds a value and its type.
    /// </summary>
    public class StreamPackage
    {

        /// <summary>
        /// Initializes a new instance of <see cref="StreamPackage"/>
        /// </summary>
        /// <param name="transportPackage">The source transport package</param>
        internal StreamPackage(TransportPackage transportPackage)
        {
            this.Type = transportPackage.Type;
            this.Value = transportPackage.Value;
            this.KafkaMessage = transportPackage.KafkaMessage;
        }

        /// <summary>
        /// Initializes a new instance of <see cref="StreamPackage"/>
        /// </summary>
        /// <param name="type">The type of the content value</param>
        /// <param name="value">The content value of the package</param>
        public StreamPackage(Type type, object value)
        {
            this.Type = type;
            this.Value = value;
        }

        /// <summary>
        /// Type of the content value
        /// </summary>
        public Type Type { get; }

        /// <summary>
        /// Content value of the package
        /// </summary>
        public object Value { get; }
        
        /// <summary>
        /// The Kafka message this stream package derives from
        /// Can be null if not consumed from broker
        /// </summary>
        public KafkaMessage KafkaMessage { get; }

        /// <summary>
        /// Serialize the package into Json
        /// </summary>
        /// <returns></returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this.Value);
        }

    }
}